src/components/refactor/dialogs/voltage-level-creation/voltage-level-creation-dialog.js

```diff
@@ -0,0 +1,211 @@
+ * @param studyUuid the study we are currently working on
+ * @param editData the data to edit
+ * @param onCreateVoltageLevel to create voltage level from other forms,
+ * @param dialogProps props that are forwarded to the generic ModificationDialog component
+ */
+
+const emptyFormData = {
+    [EQUIPMENT_ID]: '',
+    [EQUIPMENT_NAME]: '',
+    [NOMINAL_VOLTAGE]: '',
+    [SUBSTATION_ID]: '',
+    [BUS_BAR_SECTIONS]: [],
+    [BUS_BAR_CONNECTIONS]: [],
+};
+
+const schema = yup.object().shape({
+    [EQUIPMENT_ID]: yup.string().required(),
+    [EQUIPMENT_NAME]: yup.string(),
+    [NOMINAL_VOLTAGE]: yup.string().required(),
+    [SUBSTATION_ID]: yup.string().required(),
+    [BUS_BAR_SECTIONS]: yup
+        .array()
+        .of(
+            yup.object().shape({
+                [ID]: yup.string().required(),
+                [NAME]: yup.string(),
+                [HORIZONTAL_POSITION]: yup.number().min(0).required(),
+                [VERTICAL_POSITION]: yup.number().min(0).required(),
+            })
+        )
+        .min(1, 'EmptyList/bbs')
+        .test('unique-positions', (values) =>
+            controleUniqueHorizontalVertical(values, 'SameHorizAndVertPos')
+        )
+        .test('unique-ids', (values) =>
+            controleUniqueId(values, 'DuplicateId')
+        ),
+    [BUS_BAR_CONNECTIONS]: yup
+        .array()
+        .of(
+            yup.object().shape({
+                [FROM_BBS]: yup.string().nullable().required(),
+                [TO_BBS]: yup.string().nullable().required(),
+                [SWITCH_KIND]: yup.string().nullable().required(),
+            })
+        )
+        .test('bus-bar-section-link', (values) =>
+            controleBusBarSectionLink(values, 'DisconnectorBetweenSameBusbar')
+        ),
+});
+
+const VoltageLevelCreationDialog = ({
+    editData,
+    currentNode,
+    studyUuid,
+    onCreateVoltageLevel = createVoltageLevel,
+    ...dialogProps
+}) => {
+    const currentNodeUuid = currentNode?.id;
+    const { snackError } = useSnackMessage();
+    const equipmentPath = 'voltage-levels';
+
+    const methods = useForm({
+        defaultValues: emptyFormData,
+        resolver: yupResolver(schema),
+    });
+
+    const {
+        reset,
+        formState: { errors },
+    } = methods;
+
+    const fromExternalDataToFormValues = useCallback(
+        (voltageLevel, fromCopy = true) => {
+            reset({
+                [EQUIPMENT_ID]:
+                    (voltageLevel?.equipmentId ?? voltageLevel?.id) +
+                    (fromCopy ? '(1)' : ''),
+                [EQUIPMENT_NAME]:
+                    voltageLevel?.equipmentName ?? voltageLevel?.name,
+                [NOMINAL_VOLTAGE]: voltageLevel.nominalVoltage,
+                [SUBSTATION_ID]: voltageLevel.substationId,
+                [BUS_BAR_SECTIONS]: voltageLevel?.busbarSections ?? [],
+                [BUS_BAR_CONNECTIONS]: voltageLevel?.busbarConnections ?? [],
+            });
+        },
```

Comment:
use "keepDirty" param, otherwise, you can't validate form after copying values from another voltage level